### DISCUSSION TITLE
UrlSlug & Versions | Argument must be of type ?int, string given

### DISCUSSION THREAD

**Discussion by twin-elements** [pimcore/pimcore#17262] [Q&A / Support]
From the database, it gets values of string type instead of int.

An exception has been thrown during the rendering of a template ("Pimcore\Model\DataObject\Data\UrlSlug::__construct(): Argument #2 ($siteId) must be of type ?int, string given, called in /vendor/pimcore/pimcore/models/DataObject/Data/UrlSlug.php on line 184").

vendor/pimcore/pimcore/models/DataObject/Data/UrlSlug.php(184): Pimcore\Model\DataObject\Data\UrlSlug->__construct('/grochowska', '0') - **site id is a string instead od int**

The same thing happens when retrieving the last version of an object - $versionId- is a string

https://github.com/pimcore/pimcore/blob/19ccfdf6004e8230300a3760a921179a08a7cc41/models/Element/Traits/VersionDaoTrait.php#L30

Can anyone know what the problem might be?

Database serwer MariaDb 10.6.17

**Comment by kingjia90** [comment]
I believe this could be converted to an issue.
It could be some missing typecasting since the refactoring in pimcore 11, could you please check the caller and if possible even provide a PR for the fixes? Thank you in advance

**Reply by twin-elements** [reply]
No problem. I'll do it.
